rfd/headless-login.md

```diff
@@ -0,0 +1,127 @@
+2. Performing `tsh scp` from one Teleport Node to another Teleport Node.
+
+In both cases, the user needs to 1) log in with MFA on the remote host and 2) pass Per-session MFA check to connect to Teleport Node from remote host.
+
+## Details
+
+### UX
+
+The headless flow will begin when a user is connected to a remote host and performs `tsh login --headless`. At this point, nothing is assumed of the client's current environment or local login session. The user will be prompted to complete the login locally, through `tsh headless --accept` or through their web browser, with a server generated headless session ID.
+
+Once the user completes the headless login locally, `tsh login --headless` will retrieve headless certificates from the server for succeeding requests.
+
+```console
+$ tsh login --headless --proxy=proxy --user=user
+Complete headless authentication on your local device or web browser:
+> tsh headless --proxy=proxy.example.com --user=dev --accept=<session_id>
+  OR
+https://proxy.example.com/headless/<session_id>/accept
+# Wait for user action
+> Profile URL:        https://proxy.example.com:3080
+  Logged in as:       dev
+  Cluster:            root-cluster
+  Roles:              headless
+  ...
+```
+
+#### Local login
+
+If the user is not yet logged in locally, they will be prompted to login as normal through their terminal or browser. The MFA verification used to login will also be used to accept the headless login request, unblocking the `tsh login --headless` request above.
+
+```console
+$ tsh headless accept <session_id> --proxy=proxy.example.com --user=dev
+Enter password for Teleport user dev:
+Tap your YubiKey
+> Profile URL:        https://proxy.example.com
+  Logged in as:       dev
+  ...
+```
+
+If the user already has a valid login session locally in their terminal or browser, they will instead be prompted to complete a single MFA check.
+
+```console
+$ tsh headless accept <session_id> --proxy=proxy.example.com --user=dev
+Tap your YubiKey
+```
+
+#### Headless login session
+
+In a headless login session, `tsh` will request new MFA-verified certificates for every request, following the same headless login flow above to complete MFA verification.
+
+```console
+$ tsh ssh user@server01
+Complete headless authentication on your local device or web browser:
+> tsh headless accept <session_id> --request=<request_id> --proxy=proxy.example.com --user=dev 
+  OR
+https://proxy.example.com/headless/<session_id>/request/<request_id>
+# Wait for user action 
+<user@server01> $
+```
+
+To improve UX, we will also offer an option for a user to poll for and accept headless requests. This will reduce the friction of copy-pasting for every headless request.
+
+For `tsh`, we will add a new command, `tsh headless poll`, to poll for new requests for a specific headless session. When a request is found, its details will be shared in the user's local terminal. They will be prompted to accept it (`y/N`) and then verify with MFA.
```

Comment:
I like the idea of making UX easier, but I think this will be very hard to make right in CLI. Why don't we go with Teleport connect right away here? It can turn on polling automatically once it detects you are using the feature and present requests right away.

Reply:
I am CLI biased so I personally would find this flow the be more efficient if I were to use this feature, but I agree it's not ideal UX, so I don't mind scrapping it if you'd like.

> It can turn on polling automatically once it detects you are using the feature and present requests right away.

Do you mean that if you connect to the remote machine from Teleport Connect (`ssh server` from Teleport connect terminal), then initate `tsh --headless`, Teleport Connect would be able to detect that and prompt the user to authenticate it with their local Teleport Connect credentials? That's a very cool use case for Teleport Connect if so.

Reply:
@Joerger not quite. I meant that if user will use `teleport connect` to open the confirmation link, Teleport connect can set up a watch for user automatically and prompt for MFA afterwards.